plugins/lookup/tss.py

```diff
@@ -0,0 +1,117 @@
+# python 3 headers, required if submitting to Ansible
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+lookup: tss
+author: Adam Migus (adam@migus.org)
+short_description: Get secrets from Thycotic Secret Server
+description:
+    - Uses the Thycotic Secret Server Python SDK to get Secrets from Secret
+      Server using token authentication with `username` and `password` on
+      the REST API at `base_url`.
+requirements:
+    - python-tss-sdk - https://pypi.org/project/python-tss-sdk/
+options:
+    _terms:
+        description: the integer ID of the secret
+        required: True
+        type: integer
+    base_url:
+        description: the base URL of the server e.g. https://localhost/SecretServer
+        env:
+            - name: TSS_BASE_URL
+        ini:
+            - section: tss_lookup
+              key: base_url
+        required: True
+    username:
+        description: the username with which to request the OAuth2 Access Grant
+        env:
+            - name: TSS_USERNAME
+        ini:
+            - section: tss_lookup
+              key: username
+        required: True
+    password:
+        description: the password associated with the supplied username
+        env:
+            - name: TSS_PASSWORD
+        ini:
+            - section: tss_lookup
+              key: password
+        required: True
+    api_path_uri:
+        default: /api/v1
+        description: the path to append to the base URL to form a valid REST
+            API request
+        env:
+            - name: TSS_API_PATH_URI
+        required: False
+    token_path_uri:
+        default: /oauth2/token
+        description: the path to append to the base URL to form a valid OAuth2
+            Access Grant request
+        env:
+            - name: TSS_TOKEN_PATH_URI
+        required: False"""
+
+RETURN = r"""
+_list:
+    description:
+        - The JSON responses to `GET /secrets/{id}`
+        - See https://updates.thycotic.net/secretserver/restapiguide/TokenAuth/#operation--secrets--id--get>"""
+
+EXAMPLES = r"""
+- hosts: localhost
+  vars:
+      secret: "{{ lookup('tss', 1) }}"
+  tasks:
+      - debug: msg="the password is {{ (secret['items'] | items2dict(key_name='slug', value_name='itemValue'))['password'] }}"
```

Comment:
While seeing this: wouldn't it be useful if the plugin already does this transform? Or is there a situation where this transform is not wanted?

Reply:
I wondered the same thing while developing it.  The reason I went with a no transformation approach was to avoid losing a bunch of other data that's embedded in the secret.  Most of the time the developer wants this type of transformation but I didn't want to limit them to it.

Reply:
You could a boolean option (with default value `true`) whether to do this transformation. That way most users directly get the value they want, and the ones who need more can simply specify that option.

Reply:
So you're suggesting lookup('tss', 1, transform=True)

I thought that having plugins return different outputs was discouraged.  Did I misunderstand?

Reply:
No, you are right, that's not encouraged. So let's keep it this way.